### PR TITLE
GEOMESA-1767 Remote Transforms for HBase

### PR DESCRIPTION
**Pull request by spandanagrawal**
This pull request implements Remote Transforms for HBase.

All tests for the HBase Datastore have passed.

Signed-off-by: Spandan Agrawal <spandanagrawal@gmail.com>

### CODE REVIEW

File: geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java [hunk 1]
```diff
@@ -65,17 +78,20 @@ public ReturnCode filterKeyValue(Cell v) throws IOException {
...
+        CQLFilter(String sftString, String filter, String transform, String transformSchema) {
+            super(sftString, filter, transform, transformSchema);
         }
 
         @Override
         public ReturnCode filterKeyValue(Cell v) throws IOException {
             try {
                 log.trace("cell = {}", v);
-                reusable.setBuffer(v.getValueArray(), v.getValueOffset(), v.getValueLength());
+                if (hasTransform) {
```
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java]: i think we should revert this change here - if there are issues with filtering we can just always clone the value until we figure it out, but it doesn't make sense to have any logic concerning transforms in this method.

File: geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala [hunk 2]
```diff
@@ -37,6 +37,8 @@ class HBaseDataStoreTest extends Specification with LazyLogging {
   val cluster = new HBaseTestingUtility()
   var connection: Connection = _
 
+  System.setProperty("geomesa.scan.ranges.target", "100")
```
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala]: should we set this in the maven opts maybe? or at least revert it in the shutdown `step`

Comment (spandanagrawal) [inline geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala]: How can I revert it?

File: geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java (hunk 1 above)
Comment (spandanagrawal) [inline geomesa-hbase/geomesa-hbase-datastore/src/main/java/org/locationtech/geomesa/hbase/filters/JSimpleFeatureFilter.java]: I tried to revert this change but it broke the remote filters. So instead I cloned the value, and the tests now pass!

File: geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala (hunk 2 above)
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala]: just set it back to whatever it was to start with.

**Comment by spandanagrawal** [comment]
All requested changes have been made!

File: geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala (hunk 2 above)
Comment (elahrvivaz) [inline geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala]: i.e.
```
val originalRangeTarget = System.getProperty("geomesa.scan.ranges.target")
System.setProperty("geomesa.scan.ranges.target", "100")
...
cluster.shutdownMiniCluster()
System.setProperty("geomesa.scan.ranges.target", originalRangeTarget)
```

Comment (spandanagrawal) [inline geomesa-hbase/geomesa-hbase-datastore/src/test/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreTest.scala]: I just removed the line after talking to Jim :)

**Comment by elahrvivaz** [comment]
merged, thanks!
